aws/README.md

```diff
@@ -1,3 +1,93 @@
+resources get stored in a database with the time the terminator classes first found them and we approximate when to
+delete them from that. If the resource has a creation date timestamp use the Terminator base class. Otherwise use
+DbTerminator.
+
+Your terminator class should have the staticmethod `create`, the properties `id` and `name` (and `created_time` if
+you are using the Terminator base class), and the method `terminate`.
+
+The create method returns the credentials to create the client, the class name, the boto3 resource name to create the
+client, and a function for the client to use which will list all the given resources for that resource type.
+
+Here's an example:
+
+```python
+class Ec2Instance(Terminator):
+
+    @staticmethod
+    def create(credentials):
+
+        def get_instances(client):
+            return [i for r in client.describe_instances()['Reservations'] for i in r['Instances']]
+
+        return Terminator._create(credentials, Ec2Instance, 'ec2', get_instances)
+```
+
+The `id`, `name`, and `created_time` properties should return an appropriate value from self.instance which is
+a dictionary from the list of resources returned by the function defined in `create`:
+
+```python
+    @property
+    def id(self):
+        return self.instance['InstanceId']
+
+    @property
+    def name(self):
+        return self.instance['PrivateDnsName']
+
+    @property
+    def created_time(self):
+        return self.instance['LaunchTime']
+```
+
+The `terminate` method should use self.client to delete the resource.
+
+```python
+    def terminate(self):
+        self.client.terminate_instances(InstanceIds=[self.id])
+```
+
+If you want to test the terminator class with your own account you can use the cleanup.py script.
+
+Warning: Always use the --check (or -c) flag and the --target flag to avoid accidentally deleting wanted resources.
+
+It is safest if you can use two separate AWS accounts for running cleanup.py (for example, one you typically use
+for dev work and an emptyish/test account for the terminator classes to use), or a single emptyish/test account.
+
+Using cleanup.py with your own account:
+  - Ensure you're using Python 3.7+
+  - Modify config.yml to use your own accounts. These can be the same account if you're just using cleanup.py.
+    If you are using two separate accounts, lambda_account_id should be the account of the profile that's going
+    to assume the IAM role in the test_account_id. The test_account_id is where the terminator class(es) will run.
+  - Make sure you have a role called ansible-core-ci-test-dev that your AWS profile can assume. This role should have
+    the permissions to use your cleanup.py class.
+  - Set the environment variable AWS_PROFILE with the profile you want to use.
+  - Run the cleanup.py script for the new target you added to find any stray resources in us-east-1: |
+        $ python cleanup.py --stage dev --target Ec2Instance -v -c
```

Comment:
The Github Markdown renderer is not showing this as code. Maybe you need an empty line before the code starts?